### ISSUE TITLE
Create facades for some objects

### ISSUE THREAD

**Issue by andylowry** [RepreZen/KaiZen-OpenApi-Parser#37]
For some object types (e.g. `Schema`, `Parameter`), some developers may wish to work with object views that are specialized along some dimension, e.g. `type` or `'in` or whatever. The current swagger-2 parser defines separate POJOs for this, which forces certain applications to use casts and instance-of checks to get at certain properties. This can be extremely inconvenient.

Our approach will be to provide facades to provide specialized views of base objects. E.g. if you have a `Schema schema` you will be able to say `schema.asArraySchema()` or `schema.asIntegerSchema()` to get a type-specific view. A facade will eliminate accessors for properties that are not relevant to the particular specialiazation (e.g. only  a `StringSchema` should have methods for a `pattern` property). It can also specialize other methods, e.g. providing type-appropriate default-value accessors methods and/or provide additional convenience methods appropriate to the specialization.

**Comment by tedepstein** [comment]
@andylowry,  can we discuss this briefly before you get started? I have a couple of concerns, and some things that maybe worth looking at as alternative solutions: visitor pattern, and EMF's generated case method come to mind.

**Comment by andylowry** [comment]
@tedepstein Certainly. Visitor is essentially what I did in the doc generator, with a huge assist from xtend's dispatch methods making things far less verbose than they otherwise would be. But rather than solve that problem, I think it's much better to turn the tables and make the basic implementations of these types provide all the accessors related to properties defined in the specification right from the start, rather than treating those accessors as "visitors" for objects to which they may or may not be applicable. I think the use-cases for these objects are more likely to be in code that doesn't care about the specializations than otherwise, so I'd prefer to use facades to achieve the specializations when called for, rather than build them into the fundamental structure and then turn to visitors for the properties that have restricted scope or specialized behavior.

Remember that the classes we're defining here will be used as representations of some model, not implementations of that model, so it's far less likely that I'll be writing code that depends on, e.g., `getDefault()` returning an `Integer` when the schema's `type` is `integer` with `format` `int32`. I'll want to know that validation includes checks of conformable types for default values, because that's in the specification. But I don't want to have to choose the correct cast in order to access the value - or turn the operation into a visitor. I just want to be able to call `getDefault` and get an `Object` back.

Besides addressing (IMO) the more likely use-cases more easily, this approach results in a base API that's much more clearly aligned with the specification, since every object has methods related to every property defined in the spec. (With a handful of exceptions, e.g. not defining a `Component` object.)
